pkg/k2/constructs/template/properties.go

```diff
@@ -0,0 +1,68 @@
+package template
+
+import (
+	"github.com/klothoplatform/klotho/pkg/construct"
+	"github.com/klothoplatform/klotho/pkg/k2/constructs/template/inputs"
+	"github.com/klothoplatform/klotho/pkg/k2/constructs/template/property"
+	"gopkg.in/yaml.v3"
+)
+
+type Properties struct {
+	propertyMap property.PropertyMap
```

Comment:
Is there a need for a dedicated type for this?

Reply:
`PropertyMap` is largely functionality extracted from [`ResourceTemplate`](https://github.com/klothoplatform/klotho/blob/main/pkg/knowledgebase/resource_template.go#L313-L313) and applied more generically to a map instead of to a template's `Properties` field. Its primary use is for working with nested properties. Interestingly, it looks like it also inherits a lack of support for `seg1[seg2.seg3]` path syntax in its `GetProperty()` method.